tests/unit/test_utils.py

```diff
@@ -102,6 +104,9 @@ def test_exception_parsing():
     assert str(e) == "Message"
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32", reason="JAVA_HOME will often be incorrect on Windows"
```

Comment:
This and the XGBoost test on Windows are the only two that seem to fail if we don't install our dependencies in the same shell as the JAR creation / running of tests.

IMO we should opt to skip or xfail these tests for now on Windows, as the requirements to get these passing are rather rigid (requiring us to force reinstall several packages and forgo package caching for most tests), and the tests themselves don't seem too critical anyways:

- the `JAVA_HOME` bug on Windows doesn't seem to cause any critical failures elsewhere
- as expressed in its warnings, xgboost/dask isn't officially supported on Windows